openmls/src/treesync/mod.rs

```diff
@@ -372,6 +402,51 @@ impl TreeSync {
             None => None,
         })
     }
+
+    /// Return a reference to the leaf at the given `key_package_ref` or `None`
+    /// if the leaf is blank.
+    ///
+    /// Returns an error if the leaf is outside of the tree.
+    pub(crate) fn leaf_from_id(
+        &self,
+        key_package_ref: &KeyPackageRef,
+    ) -> Result<Option<&LeafNode>, TreeSyncError> {
+        let tsn = self.tree.leaf(self.leaf_index(key_package_ref)?)?;
+        Ok(match tsn.node() {
+            Some(node) => Some(node.as_leaf_node()?),
+            None => None,
+        })
+    }
+
+    /// Get the [`LeafIndex`] for a given [`KeyPackageRef`].
+    ///
+    /// This should go away and this tree should handle [`KeyPackageRef`] instead.
```

Comment:
If that is the case, can you file an issue, so we can keep track of it?

Reply:
I added a reference to #732